### PR TITLE
Feature/option to upload file or text in multipart form

### LINKED ISSUE #5547: [Feature] Add option in multipart form data to upload file or text
## Description

To allow using file meta data, the server supports an object representation for file types. The API pane needs to allow users to choose the file type to be able to use this object. The property for each key-value pair would then look like:
```
{
  "key": "keyName",
  "value": "valueField",
  "type": "FILE" (or "TEXT")
}
```

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description
Added option to upload file or text in multipart form

Screenshot 2021-08-11 at 11 59 50

Fixes #5547

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Cypress tests were written

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feature/option-to-upload-file-or-text-in-multipart-form 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.77 **(-0.05)** | 36.73 **(-0.04)** | 33.11 **(-0.04)** | 55.35 **(-0.05)**
 :green_circle: | app/client/src/components/designSystems/blueprint/ButtonComponent.tsx | 46.43 **(0.28)** | 37.11 **(-0.54)** | 57.14 **(3.29)** | 50 **(0.54)**
 :sparkles: :new: | **app/client/src/components/editorComponents/DropdownComponent.tsx** | **23.19** | **14.29** | **6.67** | **24.19**
 :sparkles: :new: | **app/client/src/components/editorComponents/form/fields/DynamicDropdownField.tsx** | **50** | **0** | **20** | **46.67**
 :green_circle: | app/client/src/components/editorComponents/form/fields/KeyValueFieldArray.tsx | 50.98 **(5.33)** | 29.09 **(4.6)** | 0 **(0)** | 52 **(5.33)**
 :green_circle: | app/client/src/pages/Editor/APIEditor/PostBodyData.tsx | 59.38 **(5.09)** | 50 **(21.43)** | 0 **(0)** | 61.29 **(5.41)**</details>

### CODE REVIEW

File: app/client/src/components/editorComponents/form/fields/KeyValueFieldArray.tsx [hunk 1]
```diff
@@ -13,8 +13,31 @@ import {
...
+  {
+    label: MultiPartOptionTypes.TEXT,
+    value: "TEXT",
+  },
+  {
+    label: MultiPartOptionTypes.FILE,
+    value: "FILE",
+  },
+];
+
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/form/fields/KeyValueFieldArray.tsx:40]: Move to APIEditorConstants

**Review by hetunandu — CHANGES_REQUESTED** [review]
Minor refactor

File: app/client/src/components/editorComponents/form/fields/KeyValueFieldArray.tsx (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/components/editorComponents/form/fields/KeyValueFieldArray.tsx]: All right. Fixed in new commit

**Review by hetunandu — APPROVED** [review]

**Comment by 96imranahmed** [comment]
Hey, do we know which release this has been packaged into? We just installed via the docker script, but this feature doesn't seem to be available.

**Pull request merged**